File: java2csharp/java/Samsung/MediaControl.MediaBrowser/IconCache.java
package com.samsung.android.sdk.sample.mediabrowser;


import android.graphics.Bitmap;
import android.net.Uri;
import android.util.LruCache;

public class IconCache extends LruCache<Uri, Bitmap> {

    private static IconCache instance;

    public static IconCache getInstance() {
        if (instance == null) {
            long size = Runtime.getRuntime().maxMemory() / 8;
            instance = new IconCache((int) size);
        }
        return instance;
    }

    protected IconCache(int maxSize) {
        super(maxSize);
    }

    @Override
    protected int sizeOf(Uri key, Bitmap value) {
        return value.getByteCount();
    }
}

File: java2csharp/csharp/Samsung/MediaControl.MediaBrowser/IconCache.cs
namespace com.samsung.android.sdk.sample.mediabrowser
{


	using Bitmap = android.graphics.Bitmap;
	using Uri = android.net.Uri;
	using LruCache = android.util.LruCache;

	public class IconCache : LruCache<Uri, Bitmap>
	{

		private static IconCache instance;

		public static IconCache Instance
		{
			get
			{
				if (instance == null)
				{
					long size = Runtime.Runtime.maxMemory() / 8;
					instance = new IconCache((int) size);
				}
				return instance;
			}
		}

		protected internal IconCache(int maxSize) : base(maxSize)
		{
		}

		protected internal override int sizeOf(Uri key, Bitmap value)
		{
			return value.ByteCount;
		}
	}

}